tests/baselines/reference/user/puppeteer.log

```diff
@@ -27,6 +27,8 @@ lib/FrameManager.js(127,15): error TS2503: Cannot find namespace 'Protocol'.
 lib/FrameManager.js(685,57): error TS2345: Argument of type 'string | number | Function' is not assignable to parameter of type 'number'.
   Type 'string' is not assignable to type 'number'.
 lib/FrameManager.js(773,15): error TS2503: Cannot find namespace 'Protocol'.
+lib/NetworkManager.js(82,5): error TS2322: Type '{}' is not assignable to type '{ [x: string]: string; }'.
+  Index signature is missing in type '{}'.
```

Comment:
Right. So this new error is a tad awkward and is cause by the index signature removal. Should an assignment like this be allowed? I looked at this - it looks like so:
```ts
const headers = {}
if (a) {
  headers["foo"] = "whatever";
}
doThingWithIndexableObject(headers);
```

So there's no _intent_ as far as types go there, but it worked before... I could patch it to work (allow assigning js literals to things with indexes), however the assignment is _certainly_ unsafe. I could have easily done
```ts
const headers = {}
if (a) {
  headers["foo"] = 42; // Not a string!
}
doThingWithIndexableObject(headers);
```
and have hilarious runtime issues hidden by the `any` indexer. So I'm on the fence on this one. I kinda wanna just leave it like this, since this is certainly more correct. @sandersn @RyanCavanaugh ?

Reply:
This is actually a new bug. I filed #26198 to track it.